web/course.go

```diff
@@ -94,15 +96,81 @@ func CoursePage(c *gin.Context) {
 		c.AbortWithStatus(http.StatusInternalServerError)
 		return
 	}
+
+	progs, err := dao.LoadProgressesForCourseAndUser((*tumLiveContext.Course).ID, (*tumLiveContext.User).ID)
+	if err != nil {
+		sentry.CaptureException(err)
+		log.Printf("%v", err)
+	}
+
+	progressResponses, progressStreams := prepareCourseProgressData(tumLiveContext, progs)
+
+	encoded, err := json.Marshal(progressResponses)
+	if err != nil {
+		sentry.CaptureException(err)
+		log.Printf("%v", err)
+	}
 	// in any other case assume either validated before or public course
-	err := templ.ExecuteTemplate(c.Writer, "course.gohtml", CoursePageData{IndexData: indexData, Course: *tumLiveContext.Course})
+	err = templ.ExecuteTemplate(c.Writer, "course-overview.gohtml",
+		CoursePageData{IndexData: indexData, Course: *tumLiveContext.Course, StreamsWithProgress: progressStreams, ProgressResponses: string(encoded)})
 	if err != nil {
 		sentrygin.GetHubFromContext(c).CaptureException(err)
 	}
 }
 
+func prepareCourseProgressData(tumLiveContext tools.TUMLiveContext, streamProgresses []model.StreamProgress) ([]ProgressInfo, []ProgressStream) {
```

Comment:
IMO this method is a little hard to wrap my head around (especially because of its complex signature), could you add a comment explaining what it does?

Reply:
It's gone now for good. :yum: